packages/chat/services/LlamaPlugin/LlamaPlugin.ts

```diff
@@ -0,0 +1,203 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+
+/**
+ * @constant {string | string} key for the API calls
+ * @private
+ */
+const _bam_key = 
+//@ts-ignore
+    (import.meta.env && import.meta.env.VITE_BAM_KEY)
+
+const _watsonxai_key = 
+//@ts-ignore
+    (import.meta.env && import.meta.env.VITE_WATSONXAI_KEY)
+
+class LlamaPluginAPI {
+
+
+    static async sendMessageWatsonX(API_URL, model, temperature, userPrompt, messages, message, session, eventNumber){
+    console.log("sending to WatsonX...")
+    if(!_watsonxai_key){
+        return {"reply":"Error: No Watsonx-ai API key specified, please append your key to your .env root file in order to access the Watson service"}
+    }
+    let max_tokens =  10000;
+    let user_name = "user"
+    let agent_name = "bot"
+    let initial_prompt = `[INST] <<SYS>> You are Watson, you'll answer all my questions. <</SYS>> [/INST]
+    `;
+
+    let WATSONXAPI_URL = "https://us-south.ml.cloud.ibm.com/ml/v1-beta/generation/text?version=2023-05-29"
+    let model_id = "meta-llama/llama-2-70b-chat"
+    let parameters = {
+    "decoding_method": "greedy",
+    "max_new_tokens": max_tokens,
+    "min_new_tokens": 0,
+    "stop_sequences": [user_name ,agent_name],
+    "repetition_penalty": 1
+    }
+    let project_id = "89ba2198-b548-4b26-99f9-50d171e091a2"
+    
+
+    let history = messages.map((conv)=>{return conv.origin+":"+conv.text}).join("\n")
+
+    prompt = initial_prompt;
+
+    let completePrompt = initial_prompt + '\n' + history 
+
+    let watsonxPayload = {
+        "b64_encoded_inputs": true,
+        "hapEnabled": true,
+        "hapText": initial_prompt,
+        "locale": "en",
+        "projectId": project_id,
+        "promptData":{
+            "imput": history,
+            "modelId": model_id,
+            "modelParameters": parameters
+        }
+    }
+
+    console.log(watsonxPayload)
+    console.log(_watsonxai_key)
```

Comment:
do we want to be console logging this info?

Reply:
removed, used for testing the CORS issues I mentioned before, can only know once I'm testing the component on the official carbon labs ibm URL.

Reply:
The carbon labs ibm url - the storybook link for that is `https://ai-canary.carbondesignsystem.com/?path=/docs/components-chat-test-input--docs` so not under the ibm.com domain...not sure if that will work?